### PR TITLE
DOCS: Update how to submit BYO credentials

### PR DESCRIPTION
**Pull request by tlimoncelli**
* byo-secrets.md: Updates and adds instructions on how to securely submit credentials for use by the project's automated testing.
* writing-providers.md: Adds a better checklist so you know you're done. 
* pr_test.yml: Add comments to make it easier to find sections of the YML

### CODE REVIEW

**Comment by tlimoncelli** [comment]
@cafferata I'd appreciate your proofreading.

File: documentation/writing-providers.md [hunk 1]
```diff
@@ -293,13 +308,25 @@ In the repo root, open `.goreleaser.yml` and add the provider to `Provider-speci
...
+3. Verify you're using the most recent version of anything you import.  (See [Step 13](#step-13-dependencies))
+4. Re-run the integration test one last time.
+  * (See [Step 7](#step-7-integration-test)).
+  * Post the results as a comment to your PR.
 5. Re-read the [maintainer's responsibilities](providers.md) bullet list.  By submitting a provider you agree to maintain it, respond to bugs, periodically re-run the integration test to verify nothing has broken, and if we don't hear from you for 2 months we may disable the provider.
 
 ## Step 16: After the PR is merged
 
 1. Remove the "provider-request" label from the PR.
 2. Verify that [documentation/providers.md](providers.md) no longer shows the provider as "requested"
```
Comment (cafferata) [inline documentation/writing-providers.md:331]: "Step 16: After the PR is merged" is no longer the current method. This changed with the migration to GitBook.

**Review by cafferata — CHANGES_REQUESTED** [review]
> @cafferata I'd appreciate your proofreading.

With all my love! 🤓

File: documentation/byo-secrets.md [hunk 2]
```diff
@@ -102,15 +105,60 @@ For example, the entry for CLOUDFLAREAPI looks like this:
...
+* The account/credentials should only access the test domain. Don't send your company's actual credentials and trust us to only touch the test domain. (this hasn't happened yet, thankfully!)
+* Renew the domain in a timely manner. This may be monitoring an email inbox you don't normally monitor.
+* Don't do anything that will get you in trouble with your employer, like charging it to your employer without permission. (this hasn't happend yet either, thankfully!)
+
+Now that we've covered all that...
 
+Create a new Github issue with a subject "Add PROVIDER to automated tests" where "PROVIDER" is the name of the provider. DO NOT SEND THE CREDENTIALS IN THE GITHUB ISSUE.  Write that you understand the above rules and would like to volunteer to maintain the credentials and account.
 
-# Caveats
+To securely send the credentials to the project, use this link: https://transfer.secretoverflow.com/u/tlimoncelli
```
[the remark this answers is not in the document]
Comment (tlimoncelli) [reply documentation/byo-secrets.md:160]: I'm pretty sure MarkDown makes bare URLs clickable.

File: documentation/writing-providers.md [hunk 3]
```diff
@@ -293,13 +308,25 @@ In the repo root, open `.goreleaser.yml` and add the provider to `Provider-speci
...
+  * `.goreleaser.yml` (Search for "Provider-specific changes")
+  * `documentation/SUMMARY.md`
+  * `documentation/providers.md` (the autogenerated table + the second one; make sure it is removed from the "requested" list)
+  * `documentation/providers/`PROVIDERNAME`.md`
+  * `integrationTest/providers.json`
+  * `providers/_all/all.go`
+3. Verify you're using the most recent version of anything you import.  (See [Step 13](#step-13-dependencies))
+4. Re-run the integration test one last time.
+  * (See [Step 7](#step-7-integration-test)).
+  * Post the results as a comment to your PR.
```
[the remark this answers is not in the document]
Comment (tlimoncelli) [reply documentation/writing-providers.md:320]: I don't want to overload people at this point.  They've already got so much to get working and debug.

File: documentation/byo-secrets.md [hunk 4]
```diff
@@ -102,15 +105,60 @@ For example, the entry for CLOUDFLAREAPI looks like this:
...
+
+2. On the "Secrets" tab, add the other env variables.
+
+Note: For the main project, secrets are added here: https://github.com/StackExchange/dnscontrol/settings/secrets/actions
+
+If you have forked the project, add these to the settings of that fork.
+
+Step 4. Submit this PR like any other.
+
+Github Actions should kick and and run the tests.
```
[the remark this answers is not in the document]
Comment (cafferata) [reply documentation/byo-secrets.md]: Is it true that this suggestion has remained unaddressed?

[Deeplink to: GitHub Actions should kick and and run the tests.](https://docs.dnscontrol.org/~/revisions/e7rW48CevoBtG9K7XvwW/developer-info/byo-secrets#:~:text=GitHub%20Actions%20should%20kick%20and%20and%20run%20the%20tests.)

File: documentation/byo-secrets.md [hunk 5]
```diff
@@ -102,15 +105,60 @@ For example, the entry for CLOUDFLAREAPI looks like this:
...
+
+Some rules:
+
+* The account/credentials should only access the test domain. Don't send your company's actual credentials and trust us to only touch the test domain. (this hasn't happened yet, thankfully!)
+* Renew the domain in a timely manner. This may be monitoring an email inbox you don't normally monitor.
+* Don't do anything that will get you in trouble with your employer, like charging it to your employer without permission. (this hasn't happend yet either, thankfully!)
+
+Now that we've covered all that...
 
+Create a new Github issue with a subject "Add PROVIDER to automated tests" where "PROVIDER" is the name of the provider. DO NOT SEND THE CREDENTIALS IN THE GITHUB ISSUE.  Write that you understand the above rules and would like to volunteer to maintain the credentials and account.
```
[the remark this answers is not in the document]
Comment (cafferata) [reply documentation/byo-secrets.md:162]: Is this suggestion maybe missed?

File: documentation/byo-secrets.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (cafferata) [reply documentation/byo-secrets.md]: Markdown certainly does that, unfortunately GitBook does not.

File: documentation/writing-providers.md [hunk 6]
```diff
@@ -293,13 +308,25 @@ In the repo root, open `.goreleaser.yml` and add the provider to `Provider-speci
...
+  * `.github/workflows/pr_test.yml` (The PROVIDERS list)
+  * `.goreleaser.yml` (Search for "Provider-specific changes")
+  * `documentation/SUMMARY.md`
+  * `documentation/providers.md` (the autogenerated table + the second one; make sure it is removed from the "requested" list)
+  * `documentation/providers/`PROVIDERNAME`.md`
+  * `integrationTest/providers.json`
+  * `providers/_all/all.go`
+3. Verify you're using the most recent version of anything you import.  (See [Step 13](#step-13-dependencies))
+4. Re-run the integration test one last time.
+  * (See [Step 7](#step-7-integration-test)).
```
[the remark this answers is not in the document]
Comment (cafferata) [reply documentation/writing-providers.md]: Is this suggestion maybe missed?

[Deeplink](https://docs.dnscontrol.org/~/revisions/e7rW48CevoBtG9K7XvwW/developer-info/writing-providers#:~:text=Post%20the%20results%20as%20a%20comment%20to%20your%20PR.)

File: documentation/writing-providers.md [hunk 7]
```diff
@@ -293,13 +308,25 @@ In the repo root, open `.goreleaser.yml` and add the provider to `Provider-speci
...
+2. Make sure the following files were created and/or updated:
+  * `OWNERS`
+  * `README.md`
+  * `.github/workflows/pr_test.yml` (The PROVIDERS list)
+  * `.goreleaser.yml` (Search for "Provider-specific changes")
+  * `documentation/SUMMARY.md`
+  * `documentation/providers.md` (the autogenerated table + the second one; make sure it is removed from the "requested" list)
+  * `documentation/providers/`PROVIDERNAME`.md`
+  * `integrationTest/providers.json`
+  * `providers/_all/all.go`
```
[the remark this answers is not in the document]
Comment (cafferata) [reply documentation/writing-providers.md]: Is this suggestion maybe missed?

[Deeplink](https://docs.dnscontrol.org/~/revisions/e7rW48CevoBtG9K7XvwW/developer-info/writing-providers#:~:text=and/or%20updated%3A-,OWNERS,-README.md)

**Review by cafferata — CHANGES_REQUESTED** [review]
The feedback effect already looks great! 🙏 I have one last couple of points that I think you may have missed.

File: documentation/writing-providers.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (cafferata) [reply documentation/writing-providers.md:320]: That can of course be a choice. Will you resolve this GitHub thread?

File: documentation/writing-providers.md (hunk 1 above)
Comment (cafferata) [reply documentation/writing-providers.md]: This GitHub thread can be marked as resolved. 👍

File: documentation/writing-providers.md [hunk 8]
```diff
@@ -157,40 +157,48 @@ Run the unit tests with this command:
 This is the most important kind of testing when adding a new provider.
 Integration tests use a test account and a real domain.
 
-* Edit [integrationTest/providers.json](https://github.com/StackExchange/dnscontrol/blob/master/integrationTest/providers.json): Add the `creds.json` info required for this provider.
+* Edit [integrationTest/providers.json](https://github.com/StackExchange/dnscontrol/blob/master/integrationTest/providers.json):
+  * Add the `creds.json` info required for this provider in the form of environment variables.
 
-For example, this will run the tests using BIND:
+WARNING: The integration test requires a DNS zone it can update.  All records
```
[the remark this answers is not in the document]
Comment (cafferata) [reply documentation/writing-providers.md]: This GitHub thread can be marked as resolved. 👍

File: documentation/writing-providers.md [hunk 9]
```diff
@@ -200,14 +208,21 @@ There is a potential bug in how TXT records are handled. Sadly we haven't found
...
-* Edit `documentation/providers.md`: Add the provider to the provider list.
-* Edit `documentation/SUMMARY.md`: Add the provider to the provider list.
+## Step 9: Update docs, CICD and other files
+
+* Edit `README.md`:
+  * Add the provider to the bullet list.
+* Edit `.github/workflows/pr_test.yml`
+  * Add the name of the provider to the PROVIDERS list.
+* Edit `documentation/providers.md`:
+  * Remove the provider from the `Requested providers` list (near the end of thedoc) (if needed).
```
[the remark this answers is not in the document]
Comment (cafferata) [reply documentation/writing-providers.md]: This GitHub thread can be marked as resolved. 👍

File: documentation/writing-providers.md [hunk 10]
```diff
@@ -293,13 +308,25 @@ In the repo root, open `.goreleaser.yml` and add the provider to `Provider-speci
 
 Here are some last-minute things to check before you submit your PR.
```
[the remark this answers is not in the document]
Comment (cafferata) [reply documentation/writing-providers.md]: This GitHub thread can be marked as resolved. 👍

File: documentation/byo-secrets.md [hunk 11]
```diff
@@ -113,7 +113,9 @@ The `*_DOMAIN` variable is stored as a "variable" while the others are stored as
 
 2. On the "Variables" tab, add `*_DOMAIN` with the name of a test domain. This domain must already exist in the account. The DNS records of the domain will be deleted, so please use a test domain or other disposable domain.
 
-Note: For the main project, variables are added here: https://github.com/StackExchange/dnscontrol/settings/variables/actions
+{% hint style="info" %}
+For the main project, variables are added here: [https://github.com/StackExchange/dnscontrol/settings/variables/actions](https://github.com/StackExchange/dnscontrol/settings/variables/actions)
+{% endhint %}
```
[the remark this answers is not in the document]
Comment (tlimoncelli) [reply documentation/byo-secrets.md:118]: Fixed

**Comment by tlimoncelli** [comment]
@cafferata  Thanks for all the feedback!  You have an excellent eye for detail!  Much appreciated!

**Pull request merged**